puffin/src/frame_data.rs

```diff
@@ -428,6 +447,32 @@ impl FrameData {
         write.write_u8(packed_streams.compression_kind as u8)?;
         write.write_all(&packed_streams.bytes)?;
 
+        let mut to_serialize_scopes = Vec::new();
+
+        let scopes = if write_all_scopes {
+            scope_details
+                .scopes_by_id(|all_scopes| all_scopes.keys().cloned().collect::<HashSet<ScopeId>>())
+        } else {
+            self.new_scopes.clone()
+        };
+
+        for new_scope_id in &scopes {
+            scope_details.scopes_by_id(|details| {
+                if let Some(details) = details.get(new_scope_id) {
+                    to_serialize_scopes.push(crate::scope_details::SerdeScopeDetails {
+                        scope_id: *new_scope_id,
+                        scope_name: details.scope_name.to_string(),
+                        function_name: details.dynamic_function_name.to_string(),
+                        file_path: details.dynamic_file_path.to_string(),
```

Comment:
We're serializing the shortened function names and file path here - is this intentional?

Reply:
Yes think so? Seems like the client / ui is most interested in the cleaned version of those strings rather then the raw unedited version.